### DISCUSSION TITLE
Conditionally enable/disable RelationManager table

### DISCUSSION THREAD

**Discussion by korematter** [filamentphp/filament#5859] [Help] [answered]
Hi, 

Does anyone know if there is a way to conditionally enable/disable the Relation Manager in getRelations() based on one of the form values. Something like the pseudo code below between **.

```
class DanceSetResource extends Resource
{
    protected static ?string $model = DanceSet::class;

    protected static ?string $navigationIcon = 'heroicon-o-collection';

public static function form(Form $form): Form
    {
        return `$form`
            ->schema([
                Select::make('competition_id')
                    ->relationship('competition', 'name')
                    ->reactive()
                    ->required(),
            ]);
    }

  public static function getRelations(): array
    {
        return [
            **if form competion_id == something then enable DancesRelationManager::class else disable it,**
        ];
    }
}
```

Thanks!

**Comment by jwhoami** [comment]
Check this answer if it helps [https://github.com/filamentphp/filament/discussions/5627](https://github.com/filamentphp/filament/discussions/5627)

**Comment by korematter** [comment, marked as answer]
Thanks @jwhoami ! This was helpful.  The actual fix was to override canViewForRecord() found in this discussion:

 discussion https://github.com/filamentphp/filament/discussions/2179

Regards,
Steve
